server/market/market.go

```diff
@@ -144,23 +153,42 @@ func NewMarket(ctx context.Context, mktInfo *dex.MarketInfo, storage db.DEXArchi
 		return nil, err
 	}
 
+	orderCoins, err := storage.ActiveOrderCoins(mktInfo.Base, mktInfo.Quote)
+	if err != nil {
+		return nil, err
+	}
+
+	lockedCoins := make(map[coinID]struct{})
+	lockedCoinsByOrder := make(map[order.OrderID][]coinID)
+	for oid, coinIDs := range orderCoins {
+		keys := make([]coinID, 0, len(coinIDs))
+		for i := range coinIDs {
+			key := &coinIDs[i]
+			keys = append(keys, key)
+			lockedCoins[key] = struct{}{}
+		}
+		lockedCoinsByOrder[oid] = keys
+	}
+
 	// Supplement the provided context with a cancel function. The Stop method
 	// may be used to stop the market, or the parent context itself may be
 	// canceled.
 	ctxInternal, cancel := context.WithCancel(ctx)
 
 	return &Market{
-		ctx:           ctxInternal,
-		cancel:        cancel,
-		running:       make(chan struct{}),
-		marketInfo:    mktInfo,
-		epochDuration: int64(mktInfo.EpochDuration),
-		orderRouter:   make(chan *orderUpdateSignal, 16),
-		book:          book.New(mktInfo.LotSize),
-		matcher:       matcher.New(),
-		swapper:       swapper,
-		auth:          authMgr,
-		storage:       storage,
+		ctx:                ctxInternal,
+		cancel:             cancel,
+		running:            make(chan struct{}),
+		marketInfo:         mktInfo,
+		epochDuration:      int64(mktInfo.EpochDuration),
+		orderRouter:        make(chan *orderUpdateSignal, 16),
+		book:               book.New(mktInfo.LotSize),
+		matcher:            matcher.New(),
+		swapper:            swapper,
+		auth:               authMgr,
+		storage:            storage,
+		lockedCoins:        lockedCoins,
+		lockedCoinsByOrder: lockedCoinsByOrder,
```

Comment:
In case anyone is snooping on these commits (but don't bother), note that this will have to happen at a higher level above the market so that locked coins will be tracked for all markets.